### PR TITLE
chapter 3

### PR DESCRIPTION
**Pull request by lrmi0808**
Ready

### CODE REVIEW

**Comment by lrmi0808** [comment]
Ready

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (anlvq-1058) [inline app/assets/stylesheets/static_pages.scss:3]: deleted this file if you don't use it

File: app/controllers/static_pages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (anlvq-1058) [inline app/controllers/static_pages_controller.rb:6]:
```suggestion
  def home; end

  def help; end
```

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (anlvq-1058) [inline app/helpers/application_helper.rb:3]: use i18n

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+      <div class="container">
+        <%= link_to "sample app", home_path, id: "logo" %>
```
Comment (anlvq-1058) [inline app/views/layouts/_header.html.erb:3]: use i18n for all string in sample app

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by lrmi0808** [comment]
Ready

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,14 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+before_action :set_locale
```
Comment (huongnt-2545) [inline app/controllers/application_controller.rb:3]: indentation
```suggestion
  before_action :set_locale
```

File: app/helpers/static_pages_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (huongnt-2545) [inline app/helpers/static_pages_helper.rb:1]: this file is not used, you can delete

File: app/views/layouts/_navigation_links.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<li><%= link_to t(:users), user_path %></li>
+
+<%= content_tag :li do%>
+  <%= link_to 'Swith to English',locale:en%>
```
Comment (huongnt-2545) [inline app/views/layouts/_navigation_links.html.erb:4]: - translate this text using I18n: 'Swith to English'
- 
```suggestion
  <%= link_to 'Swith to English', locale:en %>
```
- Fix the same error in line 8

File: app/views/static_pages/help.html.erb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<% provide :title, t(:help) %>
+
+<h1><%= t(:help)%></h1>
+<p>
+  <%= t(:greet3)%>
+  <a href="http://www.railstutorial.org/help">Rails
+  Tutorial help
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb:7]: I18n

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  page</a>.
+
+  <br><br>
+
+  <%= t(:greet4)%>
+  <a href="http://www.railstutorial.org/book"><em>Ruby
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb:13]: I18n

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, t(:home) %>
+
+<h1><%= t(:home) %></h1>
+<p>
+  <%= t(:greet1) %>
+  <a href = "http://www.rails tutorial.org/">Ruby on Rails Tutorial</a>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb:6]: I18n

File: config/routes.rb [hunk 11]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
+    resources :posts
```
Comment (huongnt-2545) [inline config/routes.rb:3]: this line is necessary in this pull request??

**Comment by lrmi0808** [comment]
Ready

File: app/controllers/application_controller.rb [hunk 12]
```diff
@@ -1,2 +1,14 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  before_action :set_locale
+
+  def set_locale
+    I18n.locale = params[:locale]||I18n.default_locale
```
Comment (anlvq-1058) [inline app/controllers/application_controller.rb:6]:
```suggestion
    I18n.locale = params[:locale] || I18n.default_locale
```

File: app/views/layouts/_navigation_links.html.erb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<li><%= link_to t(:users), user_path %></li>
+
+<%= content_tag :li do%>
+  <%= link_to t(:en_switch),locale:en%>
+<%= end unless current_page? (locale:en)%>
+
+<%= content_tag :li do%>
```
Comment (anlvq-1058) [inline app/views/layouts/_navigation_links.html.erb:7]:
```suggestion
<%= content_tag :li do %>
```

File: app/views/layouts/_navigation_links.html.erb [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<li><%= link_to t(:users), user_path %></li>
+
+<%= content_tag :li do%>
```
Comment (anlvq-1058) [inline app/views/layouts/_navigation_links.html.erb:3]:
```suggestion
<%= content_tag :li do %>
```

File: app/views/layouts/_navigation_links.html.erb [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+  <%= link_to t(:en_switch),locale:en%>
+<%= end unless current_page? (locale:en)%>
```
Comment (anlvq-1058) [inline app/views/layouts/_navigation_links.html.erb:5]:
```suggestion
  <%= link_to t(:en_switch),locale:en %>
<%= end unless current_page? (locale:en) %>
```

File: app/views/layouts/_navigation_links.html.erb [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
+<%= content_tag :li do%>
+  <%= link_to t(:jp_switch),locale:jp%>
+<%= end unless current_page? (locale:jp)%>
```
Comment (anlvq-1058) [inline app/views/layouts/_navigation_links.html.erb:9]:
```suggestion
  <%= link_to t(:jp_switch),locale:jp %>
<%= end unless current_page? (locale:jp) %>
```

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title(yield(:title))%></title>
```
Comment (anlvq-1058) [inline app/views/layouts/application.html.erb:4]:
```suggestion
    <title><%= full_title(yield(:title)) %></title>
```

File: config/routes.rb (hunk 11 above)
Comment (anlvq-1058) [reply config/routes.rb:3]: why you have `resources :posts` in this pull request?

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by lrmi0808** [comment]
Ready

File: config/routes.rb [hunk 18]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
+    root 'static_pages#home', as: 'home'
```
Comment (huongnt-2545) [inline config/routes.rb:3]: indent = 2 spaces
```suggestion
     root 'static_pages#home', as: 'home'
     etc.
```

**Review by huongnt-2545 — APPROVED** [review]
merged
I have 1 small comment, please fix it in the next pull request
You can merge this pull

**Pull request merged**